packages/hop-node/src/watchers/SyncWatcher.ts

```diff
@@ -674,6 +712,213 @@ class SyncWatcher extends BaseWatcher {
+  }
+
+  private async updateUnbondedTransferRootAmountsMap (destinationChainId: number) {
+    const totalAmounts = await this.calculateUnbondedTransferRootAmounts(destinationChainId)
+    const destinationChain = this.chainIdToSlug(destinationChainId)
+    this.unbondedTransferRootAmounts[destinationChain] = totalAmounts
+    this.lastCalculated[destinationChain] = Date.now()
+  }
+
+  async syncPendingAmounts () {
+    const pendingAmounts = BigNumber.from(0)
+    const chains = await this.bridge.getChainIds()
+    for (const destinationChainId of chains) {
+      const destinationChain = this.chainIdToSlug(destinationChainId)
+      if (
+        this.chainSlug === Chain.Ethereum ||
+        this.chainSlug === destinationChain
+      ) {
+        continue
+      }
+      await this.updatePendingAmountsMap(destinationChainId)
+    }
+  }
+
+  async syncUnbondedTransferRootAmounts () {
+    const chains = await this.bridge.getChainIds()
+    for (const destinationChainId of chains) {
+      const sourceChain = this.chainSlug
+      const destinationChain = this.chainIdToSlug(destinationChainId)
+      const shouldSkip = (
+        sourceChain === Chain.Ethereum ||
+        sourceChain === destinationChain ||
+        !this.hasSiblingWatcher(destinationChainId)
+      )
+      if (shouldSkip) {
+        continue
+      }
+      await this.updateUnbondedTransferRootAmountsMap(destinationChainId)
+    }
+  }
+
+  private async syncAvailableCredit () {
+    const chains = await this.bridge.getChainIds()
+    for (const destinationChainId of chains) {
+      const sourceChain = this.chainSlug
+      const destinationChain = this.chainIdToSlug(destinationChainId)
+      const shouldSkip = (
+        sourceChain === Chain.Ethereum ||
+        sourceChain === destinationChain ||
+        !this.hasSiblingWatcher(destinationChainId)
+      )
+      if (shouldSkip) {
+        continue
+      }
+      await this.updateAvailableCreditMap(destinationChainId)
+      const availableCredit = await this.getEffectiveAvailableCredit(destinationChainId)
+      this.logger.debug(`availableCredit (${this.tokenSymbol} ${sourceChain}→${destinationChain}): ${this.bridge.formatUnits(availableCredit)}`)
+    }
+  }
+
+  async getOruToL1PendingAmount () {
+    let pendingAmounts = BigNumber.from(0)
+    for (const chain of oruChains) {
+      const watcher = this.getSiblingWatcherByChainSlug(chain)
+      if (!watcher) {
+        continue
+      }
+
+      const destinationChainId = this.chainSlugToId(Chain.Ethereum)
+      const pendingAmount = await watcher.calculatePendingAmount(destinationChainId)
+      pendingAmounts = pendingAmounts.add(pendingAmount)
+    }
+
+    return pendingAmounts
+  }
+
+  async getOruToAllUnbondedTransferRootAmounts () {
+    let totalAmount = BigNumber.from(0)
+    for (const destinationChain in this.unbondedTransferRootAmounts) {
+      if (this.lastCalculated[this.lastCalculated[destinationChain]]) {
```

Comment:
I believe this is supposed to not be nested. I think it should just be `this.lastCalculated[destinationChain]`?

Reply:
nice catch, copy paste error. Fixed